### DISCUSSION TITLE
Add side effects to generated crud resolvers

### DISCUSSION THREAD

**Discussion by bmayen** [MichalLytek/typegraphql-prisma#252] [Q&A] [answered]
What is the best practice for triggering side effects from a generated crud resolver? For instance, I need to send messages to an aws sqs queue on create. I could do this using middleware but not sure if this the right approach.

**Comment by MichalLytek** [comment, marked as answer]
I think middlewares are the right solution for this. You can also try extending the generated resolver and overriding the method.

**Reply by bmayen** [reply]
If I override I need to completely reimplement that resolver's functionality though, correct? Or can I call super() to reuse what's already there and add my side effects in addition?

**Reply by MichalLytek** [reply]
`super(...args)` should work, I think I need to add this to examples or docs

**Reply by bmayen** [reply]
That would be great.
I'll try it out in the meantime and report back.

**Reply by bmayen** [reply]
```
import {Arg, Ctx, Info, Mutation, Resolver} from 'type-graphql';
import {CreateFooArgs, CreateFooResolver, Foo} from '@generated/type-graphql';
import {GraphQLResolveInfo} from "graphql";

@Resolver()
export class MyCreateFooResolver extends CreateFooResolver{
  @Mutation(() => Foo)
  override createFoo(
    @Ctx() ctx: any,
    @Info() info: GraphQLResolveInfo,
    @Arg('args') args: CreateFooArgs,
  ) {
    console.log('creating foo', args);
    return super.createFoo(ctx, info, args);
  }
}
```

When I compile this I get the following error:
```
/home/foo/node_modules/type-graphql/dist/schema/schema-generator.js:540
             throw new errors_1.CannotDetermineGraphQLTypeError("input", target.name, propertyName, parameterIndex, argName);
                   ^
 CannotDetermineGraphQLTypeError: Cannot determine GraphQL input type for argument named 'args' of 'createFoo' of 'MyCreateFooResolver' class. Is the value, that is used as its TS type or explicit type, decorated with a proper decorator or is it a proper input value?
 |     at Function.getGraphQLInputType (/home/foo/node_modules/type-graphql/dist/schema/schema-generator.js:540:19)
     at /home/foo/node_modules/type-graphql/dist/schema/schema-generator.js:456:32
     at Array.reduce (<anonymous>)
     at Function.generateHandlerArgs (/home/foo/node_modules/type-graphql/dist/schema/schema-generator.js:452:23)
     at /home/foo/node_modules/type-graphql/dist/schema/schema-generator.js:398:28
     at Array.reduce (<anonymous>)
     at Function.generateHandlerFields (/home/foo/node_modules/type-graphql/dist/schema/schema-generator.js:390:25)
     at Function.buildRootMutationType (/home/foo/node_modules/type-graphql/dist/schema/schema-generator.js:355:26)
     at Function.generateFromMetadataSync (/home/foo/node_modules/type-graphql/dist/schema/schema-generator.js:33:28)
     at buildSchemaSync (/home/foo/node_modules/type-graphql/dist/utils/buildSchema.js:20:55)
```

Not sure how to interpret this error. I tried changing the args type to FooCreateInput, but that isn't compatible with the base class method, which expects CreateFooArgs, and I get "not assignable to the same property in base type" error.

Any further guidance you could provide here would be much appreciated.

**Reply by MichalLytek** [reply]
It should be `@Args() args: CreateFooArgs,` not `@Arg('args') args: CreateFooArgs,`.

**Reply by bmayen** [reply]
Akk! That was it. Thanks so much.
This works as expected. I might still go with the middleware approach in the end, but this was really useful and I'm sure will come in handy sooner than later.
